ops/testing.py

```diff
@@ -1183,8 +1184,23 @@ def add_network(self, address: str, *,
             egress_subnets: List of egress subnets. Defaults to [cidr].
 
         Raises:
+            ModelError: If the endpoint is not a known relation name, or the
+                relation_id is incorrect or doesn't match the endpoint.
             ValueError: If address is not an IPv4 or IPv6 address.
         """
+        if endpoint is not None and endpoint not in self._meta.relations:
+            raise model.ModelError(f'{endpoint} is not a known relation')
+        if relation_id is not None:
+            if endpoint is None:
+                raise TypeError('endpoint must be set if relation_id is provided')
+            relation_name = self._backend._relation_names.get(relation_id)
+            if relation_name is None:
+                raise model.ModelError(
+                    f'relation_id {relation_id} has not been added (using add_relation)')
+            if endpoint != relation_name:
+                raise model.ModelError(
+                    f"endpoint {endpoint!r} does not correspond to relation_id {relation_name!r}")
```

Comment:
You say "relation_id" but include the "relation_name", you probably want both {relation_id} and {relation_name} here.

Reply:
Good call, done.